pkg/allowedpaths/cmd.go

```diff
@@ -0,0 +1,784 @@
+package allowedpaths
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"os"
+	"os/exec"
+	"path"
+	"path/filepath"
+	"runtime"
+)
+
+type AllowedCommand func(ctx context.Context, arg ...string) (*exec.Cmd, error)
```

Comment:
I have no idea if this should be `type AllowedCommand func` or 
```
type Command interface {
  Ctx(ctx context.Context, arg ...string) (*exec.Cmd, error)
}
```

and if it's an interface, is `ExpectedPaths()` or `Basename()` belong there

Reply:
I thought about both, and I think both would probably be fine -- this option ultimately felt a little easier/less cumbersome to use on the consuming side.

Reply:
I'm good either way. I'm pretty sure I've always found interfaces better in the long run, but not totally clear why